packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts

```diff
@@ -120,16 +120,28 @@ export class HttpRoute extends Resource implements IHttpRoute {
     this.httpApi = props.httpApi;
     this.path = props.routeKey.path;
 
-    let integration: HttpIntegration | undefined;
     const config = props.integration.bind({
       route: this,
       scope: this,
     });
-    integration = new HttpIntegration(this, `${this.node.id}-Integration`, {
+
+    let connectionId = config.connectionId;
+    if (config.connectionType === HttpConnectionType.VPC_LINK && !connectionId) {
+      if (!config.vpc) {
+        throw new Error('One of connectionId or vpc should be provided for private integration');
+      }
+
+      const vpcLink = this.httpApi.addVpcLink({ vpc: config.vpc });
```

Comment:
Can we call `addVpcLink()` from the `bind()` method of the base class `HttpProxyPrivateIntegration` instead? Feels like the right layer of responsibility and return the resultant vpcLink as the result of the bind.

You should be able to go from `HttpRouteIntegrationBindOptions` → `IHttpRoute` → `IHttpApi`.